src/ln/peer_handler.rs

```diff
@@ -734,21 +734,28 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 						}
 						continue;
 					},
-					Event::DisconnectPeer { ref node_id, ref msg } => {
-						if let Some(mut descriptor) = peers.node_id_to_descriptor.remove(node_id) {
-							if let Some(mut peer) = peers.peers.remove(&descriptor) {
-								if let Some(ref msg) = *msg {
-									peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg, 17)));
-									// This isn't guaranteed to work, but if there is enough free
-									// room in the send buffer, put the error message there...
-									Self::do_attempt_write_data(&mut descriptor, &mut peer);
+					Event::HandleError { ref node_id, ref action } => {
+						if let Some(ref action) = *action {
+							match *action {
+								msgs::ErrorAction::DisconnectPeer { ref msg } => {
```

Comment:
Hmm, can we DRY the ErrorAction handling here and in do_read_event? That way we don't have only DisconnectPeer handled here.